openmetadata-service/src/main/java/org/openmetadata/service/util/ElasticSearchClientUtils.java

```diff
@@ -72,25 +61,14 @@ public static RestHighLevelClient createElasticSearchClient(ElasticSearchConfigu
     }
   }
 
-  private static SSLContext createSSLContext(ElasticSearchConfiguration elasticSearchConfiguration)
+  private static SSLContext createElasticSearchSSLContext(ElasticSearchConfiguration elasticSearchConfiguration)
       throws KeyStoreException {
 
-    if (elasticSearchConfiguration.getScheme().equals("https")
-        && elasticSearchConfiguration.getTruststorePath() != null
-        && !elasticSearchConfiguration.getTruststorePath().isEmpty()) {
-      Path trustStorePath = Paths.get(elasticSearchConfiguration.getTruststorePath());
-      KeyStore truststore = KeyStore.getInstance("jks");
-      try (InputStream is = Files.newInputStream(trustStorePath)) {
-        truststore.load(is, elasticSearchConfiguration.getTruststorePassword().toCharArray());
-        SSLContextBuilder sslBuilder = SSLContexts.custom().loadTrustMaterial(truststore, null);
-        return sslBuilder.build();
-      } catch (IOException
-          | NoSuchAlgorithmException
-          | CertificateException
-          | KeyStoreException
-          | KeyManagementException e) {
-        throw new RuntimeException("Failed to crete SSLContext to for ElasticSearch Client", e);
-      }
+    if (elasticSearchConfiguration.getScheme().equals("https")) {
+      return SSLUtil.createSSLContext(
+          elasticSearchConfiguration.getTruststorePath(),
```

Comment:
@pmbrull you can use https without providing the trust store. If they do this will fail. Please revert to the older code.